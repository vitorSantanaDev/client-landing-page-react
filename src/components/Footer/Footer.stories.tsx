import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import Footer from '.'

export default {
  title: 'Footer',
  component: Footer,
  args: {
    children:
      '<p><a href="https://github.com/vitorSantanaDev">Feito com ❤️ por Vitor Santana</a></p>'
  }
} as ComponentMeta<typeof Footer>

export const Default: ComponentStory<typeof Footer> = (args) => (
  <Footer {...args} />
)
