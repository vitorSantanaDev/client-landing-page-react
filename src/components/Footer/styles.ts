import styled, { css } from 'styled-components'
import { Wrapper as TextComponent } from '../../components/TextComponent/styles'
import { Wrapper as SectionContainer } from '../../components/SectionContainer/styles'

export const Wrapper = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    padding: ${theme.spacings.small};

    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
    a {
      color: inherit;
      text-decoration: none;
    }
  `}
`
