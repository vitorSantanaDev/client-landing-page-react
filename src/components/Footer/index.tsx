import React from 'react'
import SectionContainer from '../SectionContainer'
import TextComponent from '../TextComponent'
import * as S from './styles'

interface IFooter {
  html: string
}

const Footer: React.FC<IFooter> = ({ html }) => (
  <S.Wrapper>
    <SectionContainer>
      <TextComponent>{html}</TextComponent>
    </SectionContainer>
  </S.Wrapper>
)

export default Footer
