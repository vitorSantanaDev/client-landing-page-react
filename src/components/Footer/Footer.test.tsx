import React from 'react'
import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'
import Footer from '.'

describe('<Footer />', () => {
  it('should render footer', () => {
    renderTheme(<Footer html="<h2>Footer<h2>" />)
    const heading = screen.getByRole('heading', { name: /footer/i })
    expect(heading).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderTheme(<Footer html="<h2>Footer<h2>" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
