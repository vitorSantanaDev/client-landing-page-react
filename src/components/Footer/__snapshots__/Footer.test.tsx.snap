// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
.c2 {
  width: 100%;
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
}

.c4 {
  font-size: 2.4rem;
}

.c0 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
  padding: 1.6rem;
}

.c0 .c3 {
  font-size: 1.6rem;
}

.c0 .c1 {
  padding-top: 0;
  padding-bottom: 0;
}

.c0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<footer
  class="c0"
>
  <div
    class="c1 c2"
  >
    <div
      class="c3 c4"
    >
      <h2>
        Footer
      </h2>
      <h2 />
    </div>
  </div>
</footer>
`;
